db/log_writer.cc

```diff
@@ -89,8 +93,15 @@ Status Writer::EmitPhysicalRecord(RecordType t, const char* ptr, size_t n) {
 
   // Compute the crc of the record type and the payload.
   uint32_t crc = crc32c::Extend(type_crc_[t], ptr, n);
-  crc = crc32c::Mask(crc);                 // Adjust for storage
-  EncodeFixed32(buf, crc);
+  if (db_options_->recycle_log_files) {
```

Comment:
It could, but there were only a handful of changes needed to existing tests.  And we'd probably want to add this again the next time we have some option around log::Writer (like using a stronger CRC, or direct_io, or aio, or...)